### ISSUE TITLE
IP Webcam App by Pavel Khlebovich not working

### ISSUE THREAD

**Issue by jannnfe** [KhaosT/homebridge-camera-ffmpeg#114]
Hey,
i want to use my old Android phone to stream a live video to my home app.
In the Wiki Tested Configuration this App should work: [LINK](https://github.com/KhaosT/homebridge-camera-ffmpeg/wiki/Tested-Configuration#ip-webcam-app-by-pavel-khlebovich)
Its not working for me. In the web interface of the app everything is working fine.
I can add the camera in home app but i can not view the stream.
Hope anybody can help me.
Thanks!

**Comment by orrpan** [comment]
adding this to the config worked for me

```
            "source":"**-re -i** http://src/video",
            "stillImageSource":"**-i** http://src/shot.jpg",
```

**Comment by jannnfe** [comment]
Nice thank you. Now its working! 👍

**Comment by orrpan** [comment]
Isn’t your camera on your own network? The use the lan IP, eg. 192.168.0.2:8080. No need to open ports

**Comment by jannnfe** [comment]
No its not working.
When i start Homebridge the console says:
`Kamera Garage is running on port 37594.`

So I can't connect to my camera using the Apple Home App without open this port.
But every time i interact with the camera the port will be changed?

Is there a way without disable firewall?
Thanks

**Comment by jannnfe** [comment]
I am using homebridge on a Linux ubuntu server.
On this server i opend port for Homebridge and other plugins so its working.
But for this plugin its not working because the port of the camera changed every startup or change.

My config is
[ [Android phone running IP Webcam App] -> [Linux Ubuntu Server running homebridge] -> [iPad 4 as home hub] ]->[Router] -> [Internet] -> [iPhone]
and its not working with firewall enabled and opend port 8080 on my Ubuntu server.

in this state im not able to connect to the camera or i can not add the camera to my Home App. Its loading and loading and loading...

**Comment by orrpan** [comment]
okey, my comment with port 8080 was only an example. In the camera app there a setting "Local broadcasting" where you can select a port, and it should only use that one. If it change anyway I have no clue what you should do. I set mine for 1599 on two devices and they work fine.

Also added username and password in app, this is the update in the config
```
"source":"-re -i http://<username>:<password>@<ip>:<port>/video",
"stillImageSource":"-i http://<username>:<password>@<ip>:<port>/shot.jpg",
```

**Comment by gijoecool** [comment]
I think @jannnfe means the port of the homebridge connection is always changing.

**Comment by orrpan** [comment]
@gijoecool maybe, but isn't that standard/required to setup in the homebridge config?

**Comment by gijoecool** [comment]
@orrphan, sort of. I misspoke. homebridge-camera-ffmpeg opens separate ports to stream the video to the iOS device. These ports are in addition to the homebridge port specified in the config.json. I don’t think these ports for the camera stream are configurable.

**Comment by jannnfe** [comment]
That's what I mean. Sorry for my expression. I'm all new to learning and must still find myself everywhere.

**Comment by orrpan** [comment]
@gijoecool okey didn't know that, tested now and see what you mean. My RPi2 probably don't have much of a firewall to worry about 😄

**Comment by kishankhakhar** [comment]
Is this the source code for Pavel Khlebovich app IPWebcam or am I in the wrong place?

**Comment by jannnfe** [comment]
What do you mean with "is this the source code"... What?

**Comment by jannnfe** [comment]
Port Issue is fixed here: nfarina/homebridge#927

**Comment by tinshade** [comment]
Is the source code for IP WebCam available? If so, then please link it here.
